pkg/docker/config/config.go

```diff
@@ -606,6 +634,30 @@ func findAuthentication(registry, path string, legacyFormat bool) (types.DockerA
 	return types.DockerAuthConfig{}, nil
 }
 
+func findAuthenticationForRef(auths dockerConfigFile, ref reference.Named) (types.DockerAuthConfig, error) {
+	for _, registry := range []string{ // example: quay.io/repo/image:tag
+		reference.TrimNamed(ref).String(), // quay.io/repo/image
+		stripImageName(ref),               // quay.io/repo
+		reference.Domain(ref),             // quay.io (backwards compatibility)
+	} {
```

Comment:
I think we need a dedicated (for easier unit testing) "walking" function to create all candidates.  The paths can be nested deeper, e.g. `foo.com/1/2/3/4/5/image:tag`.

Reply:
Done, added a dedicated function `pathsForRef` and added unit tests